thunder/dynamo/splitter.py

```diff
@@ -32,11 +33,46 @@
     from typing import Any
 
 
+class LazyInductorModule(torch.nn.Module):
+    def __init__(self, graph_module, fake_mode):
+        super().__init__()
+        self.graph_module = graph_module
+        self.compiled_fn = None
+        self.fake_mode = fake_mode
+
+        # For ease of debugging, we add graph attribute so GraphModule.print_readable will print it
+        self.graph = graph_module.graph
+
+    def forward(self, *args):
+        if self.compiled_fn is None:
+
+            def fake_increment_toplevel(*args, **kwargs):
+                # In PyTorch before 2.8.0, FXGraphCache assumes that it is run behind Dynamo and tries to update metrics_context.
+                # See https://github.com/pytorch/pytorch/pull/150423
+                metrics_context = torch._dynamo.utils.get_metrics_context()
+                assert not metrics_context.in_progress()
+                return
+
+            with unittest.mock.patch.object(
+                torch._dynamo.utils.CompileEventLogger, "increment_toplevel", fake_increment_toplevel
+            ):
```

Comment:
What do you think about patching only for PyTorch version <= 2.8 and adding a TODO to remove this optional patching once 2.8 is obselete?